pyanaconda/modules/subscription/installation.py

```diff
@@ -135,3 +137,138 @@ def run(self):
+        :param bool transfer_subscription_tokens: if True attempt to transfer subscription
+                                                  tokens to target system (we always transfer
+                                                  system purpose data unconditionally)
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._transfer_subscription_tokens = transfer_subscription_tokens
+
+    @property
+    def name(self):
+        return "Transfer subscription tokens to target system"
+
+    def _copy_pem_files(self, input_folder, output_folder, not_empty=True):
+        """Copy all pem files from input_folder to output_folder.
+        Files with the pem extension are generally encryption keys and certificates.
+        If output_folder does not exist, it & any parts of its path will
+        be created.
+        :param str input_folder: input folder for the pem files
+        :param str output_folder: output folder where to copy the pem files
+        :return: False if the input directory does not exists or is empty,
+                True after all pem files have be successfully copied
+        :rtype: bool
+        """
+        # check the input folder exists
+        if not os.path.isdir(input_folder):
+            return False
+        # optionally check the input folder is not empty
+        if not_empty and not os.listdir(input_folder):
+            return False
+        # create the output folder path if it does not exist
+        if not os.path.isdir(output_folder):
+            util.mkdirChain(output_folder)
+        # transfer all the pem files in the input folder
+        for pem_file_path in glob.glob(os.path.join(input_folder, "*.pem")):
+            shutil.copy(pem_file_path, output_folder)
+        # if we got this far the pem copy operation was a success
+        return True
+
+    def _copy_file_to_path(self, file_path, target_file_path):
+        if not os.path.isfile(file_path):
+            return False
+        if not os.path.isdir(os.path.dirname(target_file_path)):
+            util.mkdirChain(os.path.dirname(target_file_path))
+        shutil.copy(file_path, target_file_path)
+        return True
+
+    def _transfer_system_purpose(self):
+        # transfer the system purpose file if present
+        # - this might be needed even if the system has not been subscribed
+        #   during the installation and is therefore always attempted
+        # - this means the syspurpose tool has been called in the installation
+        #   environment & we need to transfer the results to the target system
+        if os.path.exists(self.RHSM_SYSPURPOSE_FILE_PATH):
+            log.debug("subscription: transferring syspurpose file")
+            target_syspurpose_file_path = self._sysroot + self.RHSM_SYSPURPOSE_FILE_PATH
+            self._copy_file_to_path(self.RHSM_SYSPURPOSE_FILE_PATH, target_syspurpose_file_path)
+
+    def _transfer_rhsm_config_file(self):
```

Comment:
These method can be one method with just two or three arguments.

Reply:
Yeah, lets do that. :)